modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1916,13 +1918,114 @@ def sort_by(
             Column label(s) to use to determine lexicographical ordering.
         ascending : boolean, default: True
             Whether to sort in ascending or descending order.
+        **kwargs : dict
+            Keyword arguments to pass when sorting partitions.
 
         Returns
         -------
         PandasDataframe
             A new PandasDataframe sorted into lexicographical order by the specified column(s).
         """
-        pass
+        if not isinstance(columns, list):
+            columns = [columns]
+        # When we do a sort on the result of Series.value_counts, we don't rename the index until
```

Comment:
If we deleted the indexing logic in `sort_function`, the following would fail:
```python
import modin.pandas as pd; import numpy as np
df = pd.DataFrame(np.random.randint(0, 100, (1000, 100)), columns=[f"col {i}" for i in range(100)])
df['col 1'].value_counts()
```

It fails because the way our `value_counts` works is it masks, then does the `value_counts`, then does the sort, then finally renames the index. This means that during sorting, the name of the index and the name of the masked column are the same, and so pandas complains that we have an ambiguous label.

I first tried to work around it by reimplementing `Series.value_counts` to rename the index before doing the sort, but I discovered that there were other places where the Index was renamed but the rename was not propagated to the partitions, and so we would get the same ambiguous error label, so this takes care of the case when our index shares names with our columns.